text/0000-ember-new-lang.md

```diff
@@ -0,0 +1,239 @@
+
+- Captions with synchronized media (such as video subtitles)
+- Correct dictionary lookups for translations
+- Assisting search engines
+- Improving typography in certain situations
+
+Accordingly, while the primary motivation of this RFC is to address an unresolved digital accessibility issue in Ember, it is expected that a successful implementation of the proposed `--lang` flag solution will provide additional, non-accessibility-related improvements to the baseline quality of new Ember applications.
+
+## Detailed design
+
+Link to [candidate implementation](https://github.com/josephdsumner/ember-cli/compare/master...ember-new-lang-base).
+
+We have explicitly chosen `--lang` as the flag (vs `--language`) for consistency with the HTML attribute itself. 
+
+```bash
+ember new my-app --lang en-US
+# -l is also a valid alias
+```
+
+The above ember-cli command will result in the following `index.html` header change.
+
+```html
+<html lang="en-US">
+```
+
+The flag is added to relevant ember-cli help commands, such as the following:
+
+```bash
+ember help new
+ember new <app-name> <options...>
+  ...
+  --lang (String) (Default: "") Sets the base human language of the application via index.html
+    alias: -l <value>
+```
+
+### Misusage and Error Handling
+
+
+Broadly, incorrect usage of the `--lang` flag covers three use-case categories:
+
+- The flag has been specified with a programming language as the argument
+- The flag has been specified with no argument
+- The flag has been specified with an invalid language code
+
+This RFC proposes that these cases cause the build process to halt with a revelant error and help message as opposed to simply reverting to the default value and reporting a message. **The rationale for the recommendation to halt instead of just report is that in these all of these cases, the user has explicitly typed `--lang` into their CLI tool. This is an unambiguous declaration of intent by the user to use the `--lang` flag correctly.**
+
+
+
+#### Invalid Language Codes
+
+Language codes are verified against [is-language-code](https://www.npmjs.com/package/is-language-code). (see [examples of valid ISO country codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes))
+
+If an invalid language code is given such as `--lang en-UK` the indended output should be a shell error that will halt the build.
+
+```bash
+ember new my-app --lang en-UK
+Unrecognized language subtag, "uk".
+```
+
+#### No argument passed
```

Comment:
IMO, this is just a "bug" / limitation of the command line argument parser used by Ember CLI. This RFC introduces a new named argument that takes a string value. If you didn't pass a value, it should be an error. If a second named argument follows it, it should not be parsed as the value. Basically it should work the same as `--blueprint` or anything else like it.

I think it's fine to implement whatever workaround is needed, but I'm not sure it needs to be included in the RFC. It's fine to include it for additional information, I suppose, but it shouldn't be considered "normative". Once we fix any bugs or limitations in the arguments parsing, this section will become obsolete.

TL;DR I don't think the RFC should be or is propose a permeant special error handling of this particular CLI argument.

Reply:
> I think it's fine to implement whatever workaround is needed, but I'm not sure it needs to be included in the RFC.

Ya, agreed (RE: this being a bug).

Reply:
@chancancode @rwjblue I'm sure we're willing to incorporate feedback for what in ember-cli is a bug and what's intended to be there- I mostly assume that the things that are there are intended to be there unless there are code comments that indicate otherwise.